src/components/VddkUploader/hooks/useVddkBuildImage.ts

```diff
@@ -0,0 +1,30 @@
+import { useActiveNamespace, useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { isEmpty } from '@utils/helpers';
+
+import type { VddkBuild } from '../utils/types';
+import { getVddkImageBuildResponse } from '../utils/utils';
+
+export const useVddkBuildImage = (buildName: string, onChangeVddk: (value: string) => void) => {
+  const [activeNamespace] = useActiveNamespace();
+  const [vddkBuild] = useK8sWatchResource<VddkBuild>(
+    buildName
+      ? {
+          groupVersionKind: {
+            group: 'build.openshift.io',
+            kind: 'Build',
+            version: 'v1',
+          },
+          name: buildName,
+          namespace: activeNamespace,
+        }
+      : null,
+  );
+
+  if (isEmpty(vddkBuild)) return null;
+
+  const buildImageResponse = getVddkImageBuildResponse(vddkBuild?.status);
+
+  if (buildImageResponse.variant === 'success') onChangeVddk(buildImageResponse?.body);
```

Comment:
im not sure its good to prop the onchanfge to the hook instead of using an effect on the consumer - what do u think?